make fails with error: unknown command "/bin/sh" for "yq"

Steps to reproduce:
* remove all local images of "yq"
* make

Result:
```
Status: Downloaded newer image for mikefarah/yq:latest
Error: unknown command "/bin/sh" for "yq"
Run 'yq --help' for usage.
make: *** [Makefile:131: env] Error 1
```

Cause:
env.sh has a docker run "mikefarah/yq" which pulls the latest image which for some time now is version 4.x, which incorporates some changes that break compatibility, see https://mikefarah.gitbook.io/yq/upgrading-from-v3:
* entrypoint ist now /usr/bin/yq --> which is why the error above occurs
* In v3 yq had seperate commands for reading/writing/deleting and more. In v4 all these have been embedded into a single expression.

So we should simply use mikefarah/yq:3

--- a/env.sh
+++ b/env.sh
@@ -3,7 +3,7 @@
 set -e
 
 yq_shell() {
-  docker run --rm -i -v ${PWD}:/workdir mikefarah/yq /bin/sh -c "$@"
+  docker run --rm -i -v ${PWD}:/workdir mikefarah/yq:3 /bin/sh -c "$@"
 }
 
 METAL_STACK_RELEASE_VERSION=$(yq_shell "yq r inventories/group_vars/all/images.yaml 'metal_stack_release_version'")

